src/geo/ui/legends/base/img-loader-view.js

```diff
@@ -0,0 +1,78 @@
+var $ = require('jquery');
+var Backbone = require('backbone');
+var util = require('../../../../core/util');
+
+module.exports = Backbone.View.extend({
+  initialize: function (opts) {
+    if (!opts.el) { throw new Error('element is mandatory.'); }
+    if (!opts.imageClass) { throw new Error('Image class is mandatory.'); }
+
+    this._color = this.$el.data('color');
+    this._icon = this.$el.data('icon');
+
+    this._imageClass = opts.imageClass;
+    this._lastImage = {
+      url: null,
+      content: null
+    };
+  },
+
+  _loadImage: function () {
+    var self = this;
+    var isSVG = this._isSVG(this._icon);
+
+    if (this.$el.length === 0) {
+      return;
+    }
+
+    if (isSVG) {
+      this._requestImage(this._icon, function (content) {
+        var svg = content.cloneNode(true);
+        var $svg = $(svg);
+        $svg = $svg.removeAttr('xmlns:a');
+        $svg.attr('class', self._imageClass + ' js-image');
+
+        self.$el.empty().append($svg);
+
+        $svg.css('fill', self._color);
+        $svg.find('path').css('fill', 'inherit');
+      });
+    } else {
+      var $img = $('<img crossorigin="anonymous"/>');
+      $img.attr('class', self._imageClass + ' js-image');
+      $img.attr('src', this._icon + '?req=markup');
+      this.$el.empty().append($img);
+    }
+  },
+
+  _requestImage: function (url, callback) {
+    var self = this;
+    var completeUrl = url + '?req=ajax';
+
+    if (this._lastImage.url === completeUrl) {
```

Comment:
because we load the content which is *also* from the last request https://github.com/CartoDB/cartodb.js/pull/1545/files/d75d53509f790ee85330d159b20430e0394961f8#diff-57119bee5bb9bd1ded030b77a0368d16R53

we could make this a hash, storing url and content, and match against the url, but as this is mostly when changing the color of the same icon not to do the request again, it is acceptable in this case to do the request if it is different, even if we already did before